lib/spell-check-view.coffee

```diff
@@ -107,3 +111,76 @@ class SpellCheckView
     instance = @getInstance()
     misspelling = @editor.getTextInBufferRange marker.getBufferRange()
     instance.suggest args, misspelling
+
+  initializeContextMenuListeners: =>
+    atom.views.getView(@editor).addEventListener 'mousedown', @addContextMenuEntries
```

Comment:
You'll need to find a better way to detect when a context menu is opened - there are ways to open it without using the mouse (for example, Windows has a dedicated keyboard key that opens the menu).